controllers/costmanagement_controller.go

```diff
@@ -132,17 +135,33 @@ func ReflectSpec(r *CostManagementReconciler, cost *costmgmtv1alpha1.CostManagem
 		costInput.ValidateCert = costmgmtv1alpha1.DefaultValidateCert
 	}
 
-	if !reflect.DeepEqual(cost.Spec.UploadWait, cost.Status.UploadWait) {
+	cost.Status.Upload.UploadToggle = cost.Spec.Upload.UploadToggle
+	if cost.Status.Upload.UploadToggle != nil {
+		costInput.UploadToggle = *cost.Status.Upload.UploadToggle
+	} else {
+		costInput.UploadToggle = costmgmtv1alpha1.DefaultUploadToggle
+	}
+
+	if !reflect.DeepEqual(cost.Spec.Upload.UploadWait, cost.Status.Upload.UploadWait) {
 		// If data is specified in the spec it should be used
-		cost.Status.UploadWait = cost.Spec.UploadWait
+		cost.Status.Upload.UploadWait = cost.Spec.Upload.UploadWait
 	}
-	if cost.Status.UploadWait != nil {
-		costInput.UploadWait = *cost.Status.UploadWait
+	if cost.Status.Upload.UploadWait != nil {
+		costInput.UploadWait = *cost.Status.Upload.UploadWait
 	} else {
 		r := rand.New(rand.NewSource(time.Now().UnixNano()))
 		costInput.UploadWait = r.Int63() % 35
 	}
 
+	if !reflect.DeepEqual(cost.Spec.Upload.UploadCycle, cost.Status.Upload.UploadCycle) {
+		cost.Status.Upload.UploadCycle = cost.Spec.Upload.UploadCycle
+	}
+	if cost.Status.Upload.UploadCycle != nil {
+		costInput.UploadCycle = *cost.Status.Upload.UploadCycle
+	} else {
+		costInput.UploadCycle = 6
```

Comment:
Lets use the constant here:
https://github.com/project-koku/korekuta-operator-go/pull/10/files#diff-9a43762653cce0c8d43de2353831f3ddR36